src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateService.kt

```diff
@@ -95,7 +95,7 @@ class ReleaseDateService(
 
   private fun chooseDateForHardstop(actualReleaseDate: LocalDate?, conditionalReleaseDate: LocalDate): LocalDate {
     if (actualReleaseDate == null) {
-      return conditionalReleaseDate
+      return if (workingDaysService.isNonWorkingDay(conditionalReleaseDate)) 1.workingDaysBefore(conditionalReleaseDate) else conditionalReleaseDate
```

Comment:
When the CRD is not a working day, the release date will be the first working day before the CRD.